modules/package.json

```diff
@@ -0,0 +1,3 @@
+{
+    "type": "module"
+}
```

Comment:
This pattern definitely works if the `.js` extension is important for both. I take it that is the constraint you're working too, even if `.mjs` would be slightly simpler.

Reply:
Curious, what was the point of adding this almost empty package.json with no package name, no package version?

Sincerely, 
*ES Module Bundlers*

Reply:
https://nodejs.org/api/packages.html#type

> The "type" field defines the module format that Node.js uses for all .js files that have that package.json file as their nearest parent.

Reply:
Hmm, is [spec](https://nodejs.org/api/packages.html#type) meant to be interpreted like this though? 

To just drop a nested, nearly empty `package.json` in a distribution directory?

And how is an NPM package like [pkg-dir](]https://www.npmjs.com/package/pkg-dir) expected to behave now, in it's claim to:

> Find the root directory of a Node.js project or npm package

The above `pkg-dir` behavior will fail to resolve the root of the NPM package for `tslib` after an ESM bundler follows to path `./imports/index.js` declared in [`package.json#exports['.']imports`}(https://github.com/microsoft/tslib/blob/5bfaf87d1d90a2d2aeb375aa24adf25fdbc4c6da/package.json#L32) and then executes `pkg-dir` from `node_modules/tslib/imports/index.js`

It seems the case that NOT using the `.mjs` extension was a shortcut is the problem here

Reply:
> Hmm, is it best to comprehensively interpret this [spec](https://nodejs.org/api/packages.html#type) like this though? 

The spec is quite intentionally written like this as far as I know. It intentionally doesn't use something like "package.json at the root of the package directory" multiple times

> To just drop a nested, nearly empty package.json in a distribution directory, when thinking about all the nuances that go into an NPM package and bundling/transpiling packages?

Even before node has supported `.mjs`, `.cjs`, `package.json#type` and more, it was already possible to define nested `package.json` files. It was possible to put `package.json#main` there to "redirect" the request to another location. I've been using this for years already.

> And how is an NPM package like pkg-dir expected to behave now, in it's claim to:

I'm not sure if this package is implemented correctly but a lot of similar resolvers were implemented correctly in the past. All modern module bundlers etc already work just OK with those nested `package.json` files. The rule, in fact, is quite simple if you want to just grab the root of a pkg, arguably you wouldn't even have to look for `package.json` files but rather just find a directory nested directly within `node_modules` directory